public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/TimeManagement.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+import { InlineField, RadioButtonGroup, Select } from '@grafana/ui';
+
+import { AzureQueryEditorFieldProps } from '../../types';
+
+import { setDashboardTime, setTimeColumn } from './setQueryValue';
+
+export function TimeManagement({ query, onQueryChange: onChange, schema }: AzureQueryEditorFieldProps) {
+  const [defaultTimeColumns, setDefaultTimeColumns] = useState<SelectableValue[] | undefined>();
+  const [timeColumns, setTimeColumns] = useState<SelectableValue[] | undefined>();
+
+  const setDefaultColumn = useCallback((column: string) => onChange(setTimeColumn(query, column)), [query, onChange]);
+
+  useEffect(() => {
+    if (schema && query.azureLogAnalytics?.dashboardTime) {
+      let timeColumns: SelectableValue[] = [];
+      let timeColumnsSet: Set<string> = new Set();
+      const defaultColumnsMap: Map<string, SelectableValue> = new Map();
+      const db = schema.database;
+      if (db) {
+        for (const table of db.tables) {
+          const cols = table.columns.reduce<SelectableValue[]>((prev, curr, i) => {
+            if (curr.type === 'datetime') {
+              if (!table.timespanColumn || table.timespanColumn !== curr.name) {
+                prev.push({ value: curr.name, label: `${table.name} > ${curr.name}` });
+                timeColumnsSet.add(curr.name);
+              }
+            }
+            return prev;
+          }, []);
+          timeColumns = timeColumns.concat(cols);
```

Comment:
Mutating `timeColumns` that is returned from `useState` might be problematic, as these values are meant to be immutable. I'd recommend trying to refactor this to not mutate this value.

Reply:
This isn't mutating the state value, it's mutating the local variable. I'll rename for clarity.